service/src/test/java/com/fitgoal/service/LoginServiceImplTest.java

```diff
@@ -0,0 +1,76 @@
+package com.fitgoal.service;
+
+import com.fitgoal.api.LoginService;
+import com.fitgoal.api.domain.User;
+import com.fitgoal.api.domain.UserLoginData;
+import com.fitgoal.api.exceptions.IncorrectEmailOrPasswordException;
+import com.fitgoal.dao.UserDao;
+import com.fitgoal.dao.domain.UserDto;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Optional;
+
+import static com.fitgoal.service.util.TestHelper.createUserDto;
+import static com.fitgoal.service.util.TestHelper.createUserLoginData;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LoginServiceImplTest {
+
+    @Mock
+    private UserDao userDao;
+
+    private LoginService loginService;
+
+    @Before
+    public void setUp() {
+        loginService = new LoginServiceImpl(userDao);
+    }
+
+    @Test
+    public void whenLoginUserWithCorrectEmailAndPassword_thenReturnUser() {
+        UserDto testUserDto = createUserDto();
+        UserLoginData testUserLoginData = createUserLoginData();
+
+        Mockito.when(userDao.findByEmail(testUserLoginData.getEmail()))
+                .thenReturn(Optional.of(testUserDto));
+
+        User receivedUser = loginService.login(testUserLoginData);
+
+        assertThat(receivedUser).isNotNull();
+        assertThat(receivedUser.getEmail()).isEqualTo(testUserDto.getEmail());
+    }
+
+    @Test(expected = IncorrectEmailOrPasswordException.class)
+    public void whenLoginUserWithIncorrectEmail_thenThrowIncorrectEmailOrPasswordException() {
+        UserLoginData testUserLoginData = createUserLoginData();
+        testUserLoginData.setEmail("wrongEmail@gmail.com");
+
+        Mockito.when(userDao.findByEmail(testUserLoginData.getEmail()))
+                .thenReturn(Optional.empty());
+
+        loginService.login(testUserLoginData);
```

Comment:
the test is badly configured, either you use the expected in the @Test or you verify that the exception is thrown in the assertThatThrownBy. The test as it's now doesn't rich the assertThatThrownBy, the exception is thrown in line 58 and then checked by the @Test(expected=....
The best way would be to write the following :
`        assertThatThrownBy(() -> loginService.login(testUserLoginData))
            .isInstanceOf(IncorrectEmailOrPasswordException.class);`

Reply:
Done